pkg/controllers/updaterun/controller.go

```diff
@@ -31,9 +32,26 @@ import (
 	"go.goms.io/fleet/pkg/utils"
 	"go.goms.io/fleet/pkg/utils/condition"
 	"go.goms.io/fleet/pkg/utils/controller"
+	"go.goms.io/fleet/pkg/utils/controller/metrics"
 	"go.goms.io/fleet/pkg/utils/informer"
 )
 
+// updateRunMetricsStatus is the status string for update run metrics.
+type updateRunMetricsStatus string
+
+const (
+	// updateRunMetricsStatusInitializing represents the update run is initializing in the metrics.
+	updateRunMetricsStatusInitializing updateRunMetricsStatus = "initializing"
+	// updateRunMetricsStatusProgressing represents the update run is in progress in the metrics.
+	updateRunMetricsStatusProgressing updateRunMetricsStatus = "progressing"
+	// updateRunMetricsStatusStuck represents the update run is stuck waiting for a cluster to be updated in the metrics.
+	updateRunMetricsStatusStuck updateRunMetricsStatus = "stuck"
```

Comment:
what's the diff between progressing & stuck?

Is it possible to make use of the updateRun condition itself? when we introduce the new condition later, we don't need to update the metrics again.

Reply:
`stuck` status means the updateRun has been waiting for a cluster update to finish for too long. I set a threshold of 1 minute and it's adjustable. We do not have a corresponding condition in the updateRun. All updateRuns that pass initialization phase have a `progressing` condition.

Reply:
i feel it's better to improve our condition instead of reinventing the status. You still need to explain how the conditions will be changed (i mean the state machine, eg, initializing->processing->failed) to the cx. 

"stuck status means the updateRun has been waiting for a cluster update to finish for too long. " Should we also include in the CRP condition? change the reason?

Reply:
Used conditions instead.